content/sensu-go/5.4/reference/checks.md

```diff
@@ -93,7 +72,219 @@ At every execution of a check command – regardless of success or failure – t
 Sensu agent publishes the check’s result for eventual handling by the **event
 processor** (the Sensu backend).
 
-### Check token substitution
+## Check scheduling
+
+Checks are scheduled by the Sensu backend, which schedules and
+publishes check execution requests to entities via a [publish-subscribe
+model][2].
+
+### Subscriptions
+
+Checks have a defined set of subscriptions, transport
+topics to which the Sensu backend publishes check requests. Sensu entities become
+subscribers to these topics (called subscriptions) via their individual
+subscriptions attribute. In practice, subscriptions typically correspond to
+a specific role or responsibility (for example: a webserver or database).
+
+Subscriptions are powerful primitives in the monitoring context because they
+allow you to effectively monitor for specific behaviors or characteristics
+corresponding to the function being provided by a particular system. For
+example, disk capacity thresholds might be more important (or at least
+different) on a database server as opposed to a webserver; conversely, CPU
+or memory usage thresholds might be more important on a caching system than
+on a file server. Subscriptions also allow you to configure check requests for
+an entire group or subgroup of systems rather than requiring a traditional one-to-one
+mapping.
+
+To configure subscriptions for a check, use the `subscriptions` attribute to specify an array of one or more subscription names.
+Sensu schedules checks once per interval for each agent with a matching subscription.
+For example, if we have three agents configured with the `system` subscription, a check configured with the `system` subscription results in three monitoring events per interval: one check execution per agent per interval.
+In order for Sensu to execute a check, the check definition must include a subscription that matches the subscription of at least one Sensu agent.
+
+#### Round-robin checks
+
+By default, Sensu schedules checks once per interval for each agent with a matching subscription: one check execution per agent per interval.
+However, Sensu also supports executing checks only once per interval when configured with the `round_robin` check attribute.
```

Comment:
This reads a bit strangely to me. I think it should be made clear that for a given agent, the execution _period_ is:

`1 / (interval * number_of_agents)`

when round robin is enabled.

That is, round robin does not change the scheduling interval for the check, but it does change the execution period for the agents that are subscribed.

Reply:
@echlebek This is great feedback, thank you! Here's an updated version with a diagram. Let me know what you think. I'd be happy to chat synchronously if that's easier.

By default, Sensu schedules checks once per interval for each agent with a matching subscription: one check execution per agent per interval. Sensu also supports deduplicated check execution when configured with the `round_robin` check attribute. For checks with `round_robin` set to `true`, Sensu executes the check once per interval, cycling through the available agents alphabetically according to agent name.

For example, for three agents configured with the `system` subscription (agents A, B, and C), a check configured with the `system` subscription and `round_robin` set to `true` results in one monitoring event per interval, with the agent creating the event following the pattern A -> B -> C -> A -> B -> C for the first six intervals.

![round robin](https://user-images.githubusercontent.com/11339965/55986866-2557d700-5c55-11e9-940e-9edd3ccfacbd.png)

In the diagram above, you can see that the standard check is executed by agents A, B, and C every 60 seconds, while the round robin check cycles through the available agents, resulting in each agent executing the check every 180 seconds.

_PRO TIP: You can use round robin to distribute check execution workload across multiple agents when using [proxy checks](#proxy-requests)._